chia/rpc/full_node_rpc_api.py

```diff
@@ -176,6 +178,36 @@ async def get_blockchain_state(self, _: Dict[str, Any]) -> EndpointResult:
             space = await self.get_network_space(
                 {"newer_block_header_hash": newer_block_hex, "older_block_header_hash": older_block_hex}
             )
+
+            try:
```

Comment:
if you would turn this try-block into a free function instead, to just compute the average block time, all the cases where you get a `None` block record could just be an early return, which would make this code read really easily

Reply:
I split the code to a separate free function